code/espurna/light.ino

```diff
@@ -93,73 +104,87 @@ void _setCCTInputValue(unsigned char warm, unsigned char cold) {
     _light_channel[1].inputValue = constrain(cold, 0, LIGHT_MAX_VALUE);
 }
 
-void _generateBrightness() {
+void _lightApplyBrightness(unsigned char channels = lightChannels()) {
 
     double brightness = (double) _light_brightness / LIGHT_MAX_BRIGHTNESS;
 
-    // Convert RGB to RGBW(W)
-    if (_light_has_color && _light_use_white) {
+    for (unsigned char i=0; i < channels; i++) {
+        _light_channel[i].value = _light_channel[i].inputValue * brightness;
```

Comment:
Huh. I think I missed this interaction with the real values. If bound, it needs to do value = inputValue for all of the leftover channels